### PR TITLE
First pass of docs for static search, config, writing annotations.

### PR DESCRIPTION
**Pull request by bmedx**
**Description:** Adds documentation and some minor code changes to match docs / make errors more readable.

**JIRA:** PLAT-2401

### CODE REVIEW

File: docs/extensions.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+.. code-block:: python
+
+    'annotation_finder.searchers': [
+        'javascript = code_annotations.extensions.javascript:JavascriptAnnotationExtension',
+        'python = code_annotations.extensions.python:PythonAnnotationExtension',
+    ],
+
+
+Then you can simply ``pip install -e .`` from your project directory. If all goes will you should see your extension
+being loaded when you run the static annotation tool with the `-vv` or `-vvv` option. For your extension to wory you
```
Comment (pwnage101) [inline docs/extensions.rst]: Maybe you meant "For your extension to **work**..."

File: docs/getting_started.rst [hunk 2]
```diff
@@ -14,3 +14,78 @@ Dependencies can be installed via the command below.
...
+the available options.
+
+By default the annotation search will perform linting, which makes sure that any found annotations match the structure
+listed in configuration. If any issues are found the command will fail with no report written, otherwise a YAML file
+containing the results of the search will be written to your ``report_path``. Both linting and reporting features can be
+turned off via command line flags.
+
+Add more structure to your annotations
+--------------------------------------
+Annotations can be more than simple comments. They can enforce the use of choices from a fixed list, and can be grouped
```
Comment (pwnage101) [inline docs/getting_started.rst]: Nit: "Annotation can be more than simple messages."

"comments" are an overloaded term here.

File: docs/writing_annotations.rst [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+       filename: foo/bar/events.py
+       found_by: python
+       line_number: 16
+     - annotation_data: Recommendations and email marketing events
+       annotation_token: '.. reporting_consumers::'
+       filename: openedx/core/djangoapps/user_api/legacy_urls.py
+       found_by: python
+       line_number: 18
+
+With this comment:
```
Comment (pwnage101) [inline docs/writing_annotations.rst]: "But, with this comment:"

**Comment by pwnage101** [comment]
:+1:

File: docs/extensions.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+.. code-block:: python
+
+    'annotation_finder.searchers': [
+        'javascript = code_annotations.extensions.javascript:JavascriptAnnotationExtension',
+        'python = code_annotations.extensions.python:PythonAnnotationExtension',
+    ],
+
+
+Then you can simply ``pip install -e .`` from your project directory. If all goes will you should see your extension
```
Comment (doctoryes) [inline docs/extensions.rst]: `all goes well`

File: docs/extensions.rst (hunk 1 above)
Comment (doctoryes) [inline docs/extensions.rst]: Do you mean `For your extension to work, you...`?

File: docs/getting_started.rst [hunk 5]
```diff
@@ -14,3 +14,78 @@ Dependencies can be installed via the command below.
...
+to provide more context-aware information. See :doc:`configuration` and :doc:`writing_annotations` for more information
+on how to use those options.
```
Comment (doctoryes) [inline docs/getting_started.rst]: Only now, after reading this Getting Started doc, did it dawn on me that this functionality could be useful for other annotations as well! Duh...

File: docs/writing_annotations.rst [hunk 6]
```diff
@@ -0,0 +1,205 @@
+Writing Annotations
+-------------------
+
+Annotations are structured comments that can be used to mark sections of code with distinct meaning, such as TODO's,
+deprecation notices, or places where sensitive information is being handled. Code Annotations breaks down these special
+comments into two parts- the annotation token, and the annotation data.
```
Comment (doctoryes) [inline docs/writing_annotations.rst]: Nit: Two-bulleted list? With separate sub-sections below?

File: docs/writing_annotations.rst [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+The report would fail (since the first configured annotation is not the first one in the comment):
+
+.. code-block:: bash
+
+    Search failed due to linting errors!
+    1 errors:
+    ---------------------------------
+
+    foo/bar/reporting.py::17:  ".. reporting_types::" is a member of a group, but no group is started!
+
```
Comment (doctoryes) [inline docs/writing_annotations.rst]: 👍 - nice examples!

**Pull request merged**